src/evict/evict_lru.c

```diff
@@ -2054,13 +2056,20 @@ __evict_walk_tree(WT_SESSION_IMPL *session, WT_EVICT_QUEUE *queue, u_int max_ent
 
         /*
          * If the global transaction state hasn't changed since the last time we tried eviction,
-         * it's unlikely we can make progress. Similarly, if the most recent update on the page is
-         * not yet globally visible, eviction will fail. This heuristic avoids repeated attempts to
-         * evict the same page.
+         * it's unlikely we can make progress. This heuristic avoids repeated attempts to evict the
+         * same page.
          */
-        if (!__wt_page_evict_retry(session, page) ||
-          (modified && page->modify->update_txn >= conn->txn_global.last_running))
+        if (!__wt_page_evict_retry(session, page)) {
+            WT_STAT_CONN_INCR(session, cache_eviction_skip_pages_retry);
             continue;
+        } else if (modified && page->modify->update_txn >= conn->txn_global.last_running) {
+            /*
+             * FIXME: The assumption that the eviction will fail if most recent update on the page
```

Comment:
I'd prefer to have a ticket attached to this FIXME. 
If we want to keep the FIXME I'd recommend an investigation ticket where we either remove this if branch or update the comment to say something like "this check is outdated, but we're keeping it because of ..."

Could we also add a but more context about why this assumption has changed?

Reply:
Fixed in f6176b656d9a0112686d210cdf8d28783dc815e8